lib/matplotlib/widgets.py

```diff
@@ -1044,14 +1094,23 @@ def set_active(self, index):
         if self.drawon:
             self.ax.figure.canvas.draw()
 
-        if not self.eventson:
-            return
-        for cid, func in self.observers.items():
-            func(self.labels[index].get_text())
+        self._exec_callbacks(self.labels[index].get_text())
+
+    def clear(self):
+        """Deactivate any previously activated button."""
+        if self.index_selected is not None:
+            ax_facecolor = self.ax.get_facecolor()
+            self.circles[self.index_selected].set_facecolor(ax_facecolor)
+        self.value_selected = None
+        self.index_selected = None
+
+        # calling it with no label, as all buttons are being cleared
+        self._exec_callbacks(None)
```

Comment:
I would call the callback once per label (not sure why you wouldn't?)
Otherwise this breaks backcompat with code that installs callbacks that assume that the argument is always a valid label.

Reply:
I think we have a fundamental disagreement on the role of callbacks. Let me try to give you an example of why I wouldn't: imagine you have matplotlib figure a with a CheckButtons to let the user choose which subset of the 10 animals they want to see pictures of, from a given country. User can click on cat and dog, and the program would display two pictures (a cat, a dog).. Once the user wants to change the source country, the developer would need to present a clean slate for the checkboxes for the new chosen country ( need for `.clear()`). When we run clear then, the user wouldn't expect to see pictures of 10 animals in succession when they wanted to choose a new country. This is the reason `.clear()` needs to run callbacks with a special `CLEAR` flag. In our previous/original discussions on this, I didn't even believe we need to run any callbacks, but now running it with label (which was not action by user) will create invalid/unanticipated series of actions.

Reply:
My point was not about semantics, but just that someone who previously registered a callback would not have expected the callback to need to handle the None value.

On the other hand, such code is also unlikely to ever call clear() as that's a new API, so we're fine.

Reply:
thanks!